File: Practice/AllDomains/Tutorials/30DaysOfCode/Day28RegExPatternsandIntrotoDatabases.cs
/*	author:@shivkrthakur  */
using System;
using System.Collections;
using System.IO;
using System.Linq;
using System.Text.RegularExpressions;

class Solution {
    static void Main(String[] args) {
        int N = Convert.ToInt32(Console.ReadLine());
        ArrayList names = new ArrayList();

        for(int a0 = 0; a0 < N; a0++){
            string[] tokens_firstName = Console.ReadLine().Split(' ');
            string firstName = tokens_firstName[0];
            string emailID = tokens_firstName[1];

            if(Regex.IsMatch(emailID,"@gmail"))
            {
                names.Add(firstName);
            }
        }

        names.Sort();
        foreach(var item in names)
        {
            Console.WriteLine(item);
        }
    }
}

File: Practice/AllDomains/Tutorials/30DaysOfCode/Day28RegExPatternsandIntrotoDatabases.js
/*	author:@shivkrthakur  */
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function main() {
    var N = parseInt(readLine());
    var output = [];
    for(var a0 = 0; a0 < N; a0++){
        var firstName_temp = readLine().split(' ');
        var firstName = firstName_temp[0];
        var emailID = firstName_temp[1];
        var patt = new RegExp(".+@gmail.com");
        if(patt.exec(emailID)) output.push(firstName);
    }
    output.sort();
    console.log(output.join('\n'));

}
